wgpu-core/src/device/mod.rs

```diff
@@ -1197,23 +1198,22 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
             }
         }
 
-        // TODO: deduplicate the bind group layouts at some level.
-        // We can't do it right here, because in the remote scenario
-        // the client need to know if the same ID can be used, or not.
-        if false {
+        let (device_guard, mut token) = hub.devices.read(&mut token);
+        let device = &device_guard[device_id];
+
+        // If there is an equivalent BGL, just bump the refcount and return it.
+        {
             let (bgl_guard, _) = hub.bind_group_layouts.read(&mut token);
             let bind_group_layout_id = bgl_guard
                 .iter(device_id.backend())
                 .find(|(_, bgl)| bgl.entries == entry_map);
```

Comment:
Now I don't think the amount of bind group layouts to be particularly high so this can stay as is for now, but we should keep an eye on this code in profiles and stuff to make sure this linear search isn't an issue.

Reply:
BGL creation is really really cold code, so it's totally fine here